### PR TITLE
Allow alert filter autotune to absorb different feature set from user  input

### PR DESCRIPTION
**Pull request by cyy0714**
To expend the usage of alert filter autotune, the endpoint is updated to allow user to indicate the feature set to be used in the autotune.

### CODE REVIEW

File: thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/dashboard/resources/DetectionJobResource.java [hunk 1]
```diff
@@ -577,7 +578,8 @@ public Response tuneAlertFilter(@PathParam("functionId") long id,
       @QueryParam("end") String endTimeIso,
       @QueryParam("autoTuneType") @DefaultValue("AUTOTUNE") String autoTuneType,
       @QueryParam("holidayStarts") @DefaultValue("") String holidayStarts,
-      @QueryParam("holidayEnds") @DefaultValue("") String holidayEnds) {
+      @QueryParam("holidayEnds") @DefaultValue("") String holidayEnds,
+      @QueryParam("tuningFeatures") String features) {
```
Comment (cecilynie) [inline thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/dashboard/resources/DetectionJobResource.java]: Please add explanation in javadoc for features. Thanks!

File: thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/dashboard/resources/DetectionJobResource.java [hunk 2]
```diff
@@ -589,6 +591,11 @@ public Response tuneAlertFilter(@PathParam("functionId") long id,
     BaseAlertFilter alertFilter = alertFilterFactory.fromSpec(anomalyFunctionSpec.getAlertFilter());
     // create alert filter auto tune
     AutotuneConfigDTO autotuneConfig = new AutotuneConfigDTO(alertFilter);
+    if (StringUtils.isNotBlank(features)) {
+      Properties autotuneProperties = autotuneConfig.getTuningProps();
+      autotuneProperties.setProperty(AUTOTUNE_FEATURE_KEY, features);
```
Comment (cecilynie) [inline thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/dashboard/resources/DetectionJobResource.java]: Here you are changing feature set. Please log the previous feature set and your updated features.

File: thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/dashboard/resources/DetectionJobResource.java (hunk 1 above)
Comment (cyy0714) [reply thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/dashboard/resources/DetectionJobResource.java]: Added. Thanks for the comment.

Comment (cyy0714) [reply thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/dashboard/resources/DetectionJobResource.java]: Added! Thanks!

File: thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/dashboard/resources/DetectionJobResource.java (hunk 2 above)
Comment (cyy0714) [reply thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/dashboard/resources/DetectionJobResource.java]: The old feature is now logged in the properties. Thanks for the comment

**Pull request merged**